open-api/rest-catalog-open-api.yaml

```diff
@@ -3964,6 +4144,292 @@ components:
+          type: string
+        event-count:
+          type: integer
+          description: Number of events in the request / batch of events
+        timestamp-ms:
+          type: integer
+          format: int64
+          description: >
+            Timestamp when this transaction occurred (epoch milliseconds).
+            Timestamps are not guaranteed to be unique. Typically all events in
+            a transaction will have the same timestamp.
+        actor:
+          type: string
+          description: >
+            The actor who performed the operation, such as a user or service account.
+            The content of this field is implementation specific.
+        operation:
+          type: object
+          description: >
+            The operation that was performed, such as creating or updating a table.
+            Clients should discard events with unknown operation types.
+          discriminator:
+            propertyName: operation-type
+            mapping:
+              create-table: "#/components/schemas/CreateTableOperation"
+              register-table: "#/components/schemas/RegisterTableOperation"
+              drop-table: "#/components/schemas/DropTableOperation"
+              update-table: "#/components/schemas/UpdateTableOperation"
+              rename-table: "#/components/schemas/RenameTableOperation"
+              create-view: "#/components/schemas/CreateViewOperation"
+              drop-view: "#/components/schemas/DropViewOperation"
+              update-view: "#/components/schemas/UpdateViewOperation"
+              rename-view: "#/components/schemas/RenameTableOperation"
+              create-namespace: "#/components/schemas/CreateNamespaceOperation"
+              update-namespace-properties: "#/components/schemas/UpdateNamespacePropertiesOperation"
+              drop-namespace: "#/components/schemas/DropNamespaceOperation"
+              custom: "#/components/schemas/CustomOperation"
+          oneOf:
+            - $ref: "#/components/schemas/CreateTableOperation"
+            - $ref: "#/components/schemas/RegisterTableOperation"
+            - $ref: "#/components/schemas/DropTableOperation"
+            - $ref: "#/components/schemas/UpdateTableOperation"
+            - $ref: "#/components/schemas/RenameTableOperation"
+            - $ref: "#/components/schemas/CreateViewOperation"
+            - $ref: "#/components/schemas/DropViewOperation"
+            - $ref: "#/components/schemas/UpdateViewOperation"
+            - $ref: "#/components/schemas/RenameTableOperation"
+            - $ref: "#/components/schemas/CreateNamespaceOperation"
+            - $ref: "#/components/schemas/UpdateNamespacePropertiesOperation"
+            - $ref: "#/components/schemas/DropNamespaceOperation"
+            - $ref: "#/components/schemas/CustomOperation"
+
+    CreateTableOperation:
+      required:
+        - operation-type
+        - identifier
+        - table-uuid
+      properties:
```

Comment:
The `UpdateTableOperation` includes the changes. That could also be included here as optional. Why not include them?

Reply:
I am not sure if the semantic of an `Update` makes sense if we don't have a proper initial state. The `CreateTableRequest` also works without updates.

I believe that for a staged create, we also don't have a good way in our SDKs to obtain all updates, unless we cache all changes for the stage create and append updates of the commit.